rasa/core/policies/policy.py

```diff
@@ -139,8 +126,8 @@ def __init__(
     ) -> None:
         """Constructs a new Policy object."""
         if featurizer is None:
-            featurizer = _featurizer_from_policy_config(config)
-        self.__featurizer = self._create_featurizer(featurizer)
+            featurizer = self._create_featurizer(config)
+        self.__featurizer = featurizer
```

Comment:
```suggestion
            self.__featurizer = self._create_featurizer(config)
        else:
            self.__featurizer = copy.deep_copy(featurizer)
```

Reply:
Do we actually need the `copy.deepcopy`? It seemed like an unused relict to me as we are not sharing the featurizers between policies anyway 🤷🏻

Reply:
I have no clue 😅  I know you mentioned that some of those are set by `ensemble.py` but I never looked into how they do that

Reply:
Haha, I'd dump it for now as the line was super old and no test fails.